### PR TITLE
literaturesuggest: allow book and chapter submissions

### PR DESCRIPTION
**Pull request by Glignos**
Added book and chapter options for document types as well as their fields.
(Also bookautocompletion)

### CODE REVIEW

File: inspirehep/modules/literaturesuggest/tasks.py [hunk 1]
```diff
@@ -153,6 +169,37 @@ def _is_arxiv_url(url):
...
         )
 
+    if form_fields.get('type_of_doc') == 'book chapter':
+        builder.add_book(
+            publisher=form_fields.get('publisher_name'),
+            place=form_fields.get('publication_place'),
+            date=form_fields.get('publication_date'))
+        title = form_fields.get('series_title')
+        result = check_book_existance(title)
+        if(result.rowcount > 0):
```
Comment (jmartinm) [inline inspirehep/modules/literaturesuggest/tasks.py]: No need for parenthesis.

File: inspirehep/modules/records/receivers.py [hunk 2]
```diff
@@ -67,9 +67,17 @@ def enhance_record(sender, json, *args, **kwargs):
...
     populate_experiment_suggest(sender, json, *args, **kwargs)
     populate_abstract_source_suggest(sender, json, *args, **kwargs)
+    add_book_autocomplete(sender, json, *args, **kwargs)
     after_record_enhanced.send(json)
 
 
+def add_book_autocomplete(sender, json, *args, **kwargs):
+    if 'book' in force_list(json.get('document_type')):
+        result = json.get('bookautocomplete', [])
+        result.extend(force_list(get_value(json, 'titles.title')))
```
Comment (jmartinm) [inline inspirehep/modules/records/receivers.py]: Can this be added to the mapping (hep.json) as a `copy_to` ?

File: inspirehep/modules/literaturesuggest/tasks.py [hunk 3]
```diff
@@ -153,6 +169,37 @@ def _is_arxiv_url(url):
             date=form_fields.get('thesis_date')
         )
 
+    if form_fields.get('type_of_doc') == 'book chapter':
```
Comment (jmartinm) [inline inspirehep/modules/literaturesuggest/tasks.py]: Since the logic gets a bit complicated with the addition of these types of documents, add some unit tests (see https://github.com/inspirehep/inspire-next/blob/master/tests/unit/literaturesuggest/test_literaturesuggest_tasks.py#L321) where you isolate different combinations of `type_of_doc` and what you expect to get as the final record.

**Comment by jacquerie** [comment]
I'm wondering if, when this feature is finished, we should coordinate with @StellaCh before deployment it to announce it to the users.

**Comment by jacquerie** [comment]
Depends on https://github.com/inspirehep/inspire-schemas/issues/144.

**Comment by Glignos** [comment]
This is failing, because it needs this to be merged and backported https://github.com/inspirehep/inspire-schemas/pull/162

**Comment by jacquerie** [comment]
This is no longer blocked by https://github.com/inspirehep/inspire-schemas/issues/144.

**Comment by michamos** [comment]
Just tested it on https://link.springer.com/book/10.1007/978-94-007-5143-9.
It looks quite good, except the year that ends up in `publication_info`, but should end up in `imprints` for a book.

File: inspirehep/modules/literaturesuggest/static/js/literaturesuggest/data_sources/data_source.js [hunk 4]
```diff
@@ -104,6 +104,7 @@ define(function(require, exports, module) {
           url: this.url + id
         })
         .always(function(data) {
+          depositionType = data.query.type;
```
Comment (jmartinm) [inline inspirehep/modules/literaturesuggest/static/js/literaturesuggest/data_sources/data_source.js]: maybe this is not yet complete, at the moment you are making the variable global and then you are not using it?

File: inspirehep/modules/literaturesuggest/static/js/literaturesuggest/literature_submission_form.js [hunk 5]
```diff
@@ -183,6 +197,59 @@ define(function(require, exports, module) {
...
+          clear_fields(['find_book', 'parent_book']);
+          var group_id = document.getElementById("state-group-page_start")
+          group_id.removeChild(group_id.firstChild);
+          document.getElementById('state-group-find_book').style.display = 'block';
+          document.getElementById('state-group-book_title').style.display = 'block';
+        });
+      }
+
+      function addBookInfoField_success(title, authors_list, publication_date) {
+        var book_label = document.createElement("div");
```
Comment (jmartinm) [inline inspirehep/modules/literaturesuggest/static/js/literaturesuggest/literature_submission_form.js]: Same as above, using jQuery would make it more compact

File: inspirehep/modules/literaturesuggest/static/js/literaturesuggest/literature_submission_form.js [hunk 6]
```diff
@@ -183,6 +197,59 @@ define(function(require, exports, module) {
...
+        var group_elem = document.getElementById("state-group-page_start")
+        group_elem.insertBefore(book_label, group_elem.firstChild);
+        $("#books-message_success > p").replaceWith('<p> You have selected <b>' + title + '</b> \
+                            ' + authors_list + '\
+                            ' + publication_date + ' </p>');
+      };
+
+      function add_book_values(datum) {
+        if (datum.self.$ref !== undefined) {
+          datum.titles[0].title
```
Comment (jmartinm) [inline inspirehep/modules/literaturesuggest/static/js/literaturesuggest/literature_submission_form.js]: what is this line?

File: inspirehep/modules/literaturesuggest/static/js/literaturesuggest/literature_submission_form.js [hunk 7]
```diff
@@ -183,6 +197,59 @@ define(function(require, exports, module) {
...
+          document.getElementById("parent_book").value = datum.self.$ref;
+        }
+        addBookInfoField_success(datum.titles[0].title, datum.authors.map(function(a) { return a.full_name.replace(',', '') }), datum.imprints[0].date);
```
Comment (jmartinm) [inline inspirehep/modules/literaturesuggest/static/js/literaturesuggest/literature_submission_form.js]: `imprints` probably does not always exist

File: inspirehep/modules/literaturesuggest/static/js/literaturesuggest/literature_submission_form.js [hunk 8]
```diff
@@ -183,6 +197,59 @@ define(function(require, exports, module) {
...
+      }
+
+      this.$book.bind('typeahead:selected', function(obj, datum, name) {
+        add_book_values(datum);
+        create_bk_button();
+        document.getElementById('state-group-find_book').style.display = 'none';
```
Comment (jmartinm) [inline inspirehep/modules/literaturesuggest/static/js/literaturesuggest/literature_submission_form.js]: Using jQuery `$('#state-group-find_book').hide()`

File: inspirehep/modules/literaturesuggest/static/js/literaturesuggest/literature_submission_form.js [hunk 9]
```diff
@@ -267,6 +342,24 @@ define(function(require, exports, module) {
...
 
+        this.$book.on('change blur typeahead:selected', function() {
+          if (!this.$bookId.val() && $.trim(this.$book.val())) {
+            this.$bookInfoField.show();
+          } else {
+            this.$bookInfoField.hide();
+          }
+        }.bind(this));
+
+      // for spinner at conferences typeahead
```
Comment (jmartinm) [inline inspirehep/modules/literaturesuggest/static/js/literaturesuggest/literature_submission_form.js]: not conferences in this case

File: inspirehep/modules/literaturesuggest/static/js/literaturesuggest/mapper.js [hunk 10]
```diff
@@ -73,18 +73,24 @@ define(function(require, exports, module) {
...
+      if (this.special_mapping[data.type]) {
+        special_mapping = this.special_mapping[data.type](data);
       }
 
       var mapping = $.extend({}, common_mapping, special_mapping);
       if (mapping.authors) {
         mapping.authors = $.map(mapping.authors, this.extract_contributor);
       }
-
+      //mapping['type_of_doc'] = depositionType;
```
Comment (jmartinm) [inline inspirehep/modules/literaturesuggest/static/js/literaturesuggest/mapper.js]: leftover

File: inspirehep/modules/literaturesuggest/static/js/literaturesuggest/mapper.js [hunk 11]
```diff
@@ -73,18 +73,24 @@ define(function(require, exports, module) {
      * @param depositionType {String} type of deposition
      * @returns {*}
      */
-    map: function(data, depositionType) {
+    map: function(data) {
+      if (data.type === 'journal-article'){
```
Comment (jmartinm) [inline inspirehep/modules/literaturesuggest/static/js/literaturesuggest/mapper.js]: So instead of having here hardcoded values from `crossref` you did not find a way to do it in `doi.js` ?

File: inspirehep/modules/literaturesuggest/tasks.py [hunk 12]
```diff
@@ -40,18 +42,17 @@ def _is_arxiv_url(url):
 
     form_fields = copy.deepcopy(formdata)
     filter_empty_elements(
-        form_fields, ['authors', 'supervisors', 'report_numbers']
+        form_fields, ['authors', 'supervisors', 'report_numbers']  # , 'isbns']
```
Comment (jmartinm) [inline inspirehep/modules/literaturesuggest/tasks.py]: leftover

File: inspirehep/modules/literaturesuggest/tasks.py [hunk 13]
```diff
@@ -123,6 +115,33 @@ def _is_arxiv_url(url):
             date=form_fields.get('thesis_date')
         )
 
+    if form_fields.get('type_of_doc') == 'chapter':
+        if not form_fields.get('journal_title'):  # if there is a journal_title then we store it as a journal
```
Comment (jmartinm) [inline inspirehep/modules/literaturesuggest/tasks.py]: The code and the comment seem to contradict each other

File: inspirehep/modules/literaturesuggest/tasks.py [hunk 14]
```diff
@@ -123,6 +115,33 @@ def _is_arxiv_url(url):
...
+            builder.add_book_series(title=form_fields.get('series_title'))
+
+    if form_fields.get('type_of_doc') == 'book':
+            if form_fields.get('journal_title'):  # if there is a journal_title then we store it as a journal
```
Comment (jmartinm) [inline inspirehep/modules/literaturesuggest/tasks.py]: I am not sure about the comment

File: inspirehep/modules/records/receivers.py [hunk 15]
```diff
@@ -67,6 +67,25 @@ def enhance_record(sender, json, *args, **kwargs):
...
+
+def add_book_autocomplete(sender, json, *args, **kwargs):
+    if 'book' in force_list(json.get('document_type')):
+        result = json.get('bookautocomplete', [])
+        result.extend(force_list(get_value(json, 'titles.title')))
+        result.extend(force_list(get_value(json, 'authors.full_name')))
+        json['bookautocomplete'] = result
+
+
+def populate_inspire_subjects(sender, json, *args, **kwargs):
```
Comment (jmartinm) [inline inspirehep/modules/records/receivers.py]: This part seems unrelated to this PR

File: tests/integration/utils/test_normalizers.py [hunk 16]
```diff
@@ -23,6 +23,11 @@
 from __future__ import absolute_import, division, print_function
 
 from inspirehep.utils.normalizers import normalize_journal_title
+from inspirehep.modules.literaturesuggest import normalizers
+
+
+class DummyObj(object):
```
Comment (jmartinm) [inline tests/integration/utils/test_normalizers.py]: there is already a MockObj in `unit/helpers/mocks`

File: tests/integration/utils/test_normalizers.py [hunk 17]
```diff
@@ -32,3 +37,35 @@ def test_normalize_journal_title(app):
...
+        'series_title': 'Phys.Rev.',
+        'journal_title': '',
+        'type_of_doc': 'book'
+    }
+
+    expected = {'series_title': 'Phys.Rev.',
+                'journal_title': 'Phys.Rev.',
+                'type_of_doc': 'book'}
+    result = normalizers.normalize_journal_title(obj, formdata)
+    print(result)
```
Comment (jmartinm) [inline tests/integration/utils/test_normalizers.py]: leftover

File: tests/integration/utils/test_normalizers.py [hunk 18]
```diff
@@ -32,3 +37,35 @@ def test_normalize_journal_title(app):
...
+        'book_title': 'The Large Hadron Collider',
+        'parent_book': '',
+        'type_of_doc': 'chapter'
+    }
+
+    expected = {'book_title': 'The Large Hadron Collider',
+                'parent_book': "http://localhost:5000/api/literature/1373790",
+                'type_of_doc': 'chapter'}
+    result = normalizers.find_book_id(obj, formdata)
+    print(result)
```
Comment (jmartinm) [inline tests/integration/utils/test_normalizers.py]: leftover

**Comment by jmartinm** [comment]
There is a way to make the book autocompletion misbehave, see:

book

**Comment by jmartinm** [comment]
Put an extra space before the link

screen shot 2017-07-06 at 4 15 47 pm

File: inspirehep/modules/forms/static/js/forms/books_typeahead.js [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+        return data.control_number;
+      },
+      displayKey: null,
+      displayfn: function(obj) {
+        return obj;
+      },
+      dataEngine: new Bloodhound({
+        name: 'books',
+        remote: {
+          url: '/api/literature?q=bookautocomplete:%QUERY*',
```
Comment (Glignos) [inline inspirehep/modules/forms/static/js/forms/books_typeahead.js]: Right now the query supports searching in multiple fields,will it keep this behaviour with this change?

File: inspirehep/modules/forms/static/js/forms/books_typeahead.js [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+            return $.map(response.hits.hits, function(el) {
+              el.metadata.title = el.metadata.titles[0].title;
+              if(typeof el.metadata.isbns !== 'undefined'){
+                el.metadata.isbn = el.metadata.isbns[0].value;
+              }
+              if(typeof el.metadata.earliest_date !== 'undefined'){
+                el.metadata.date = el.metadata.earliest_date;
+              }(function(a) {return a.foo;});
+              if(typeof el.metadata.authors !== 'undefined'){
+                  el.metadata.author_list = el.metadata.authors.map( function(a) {return a.full_name.replace(',','')});
```
Comment (Glignos) [inline inspirehep/modules/forms/static/js/forms/books_typeahead.js]: Well since im separating the authors with commas im removing the name surname comma separator.

File: inspirehep/modules/forms/static/js/forms/books_typeahead.js (hunk 19 above)
Comment (jmartinm) [inline inspirehep/modules/forms/static/js/forms/books_typeahead.js]: It would be the same, you need to define in the ES mapping a field of type `completion` and then in a receiver (see [example](https://github.com/inspirehep/inspire-next/commit/5fd4bd4595cdc5f66add7d669f2c864e7d991720#diff-84b69cc07db1d1775867935adf05cabaR311)) populate that field.

File: inspirehep/modules/forms/static/js/forms/books_typeahead.js (hunk 20 above)
Comment (jmartinm) [inline inspirehep/modules/forms/static/js/forms/books_typeahead.js]: 👍 ok thats fine

File: tests/integration/utils/test_normalizers.py (hunk 16 above)
Comment (Glignos) [inline tests/integration/utils/test_normalizers.py]: I cant use a unit test helper in an integration test

File: inspirehep/modules/literaturesuggest/static/js/literaturesuggest/mapper.js (hunk 11 above)
Comment (Glignos) [inline inspirehep/modules/literaturesuggest/static/js/literaturesuggest/mapper.js]: No because due to the naming I cant use the 'journal-article' to name a mapping

**Comment by Glignos** [comment]
@jmartinm Can you do a quick check again? :)

File: inspirehep/modules/literaturesuggest/static/js/literaturesuggest/literature_submission_form.js [hunk 21]
```diff
@@ -183,6 +197,48 @@ define(function(require, exports, module) {
 
       conferencesTypeahead(this.$conference);
 
+      booksTypeahead(this.$book);
+
+      function clear_fields(fields) {
```
Comment (jmartinm) [inline inspirehep/modules/literaturesuggest/static/js/literaturesuggest/literature_submission_form.js]: For names of functions in JS, keep style `clearFields`

File: inspirehep/modules/records/receivers.py [hunk 22]
```diff
@@ -67,6 +67,27 @@ def enhance_record(sender, json, *args, **kwargs):
     populate_abstract_source_suggest(sender, json, *args, **kwargs)
     populate_title_suggest(sender, json, *args, **kwargs)
     after_record_enhanced.send(json)
+    add_book_autocomplete(sender, json, *args, **kwargs)
+    after_record_enhanced.send(json)
+
+
+def add_book_autocomplete(sender, json, *args, **kwargs):
+    if 'book' in force_list(json.get('document_type')):
```
Comment (jmartinm) [inline inspirehep/modules/records/receivers.py]: I don't think you need `force_list`, document type is an array in the schema

File: inspirehep/modules/records/receivers.py [hunk 23]
```diff
@@ -67,6 +67,27 @@ def enhance_record(sender, json, *args, **kwargs):
...
+        result = json.get('bookautocomplete', [])
+        result.extend(force_list(get_value(json, 'titles.title')))
+        result.extend(force_list(get_value(json, 'authors.full_name')))
+        titles = force_list(get_value(json, 'titles.title'))
```
Comment (jmartinm) [inline inspirehep/modules/records/receivers.py]: You are repeating these lines above.

File: inspirehep/modules/records/receivers.py [hunk 24]
```diff
@@ -67,6 +67,27 @@ def enhance_record(sender, json, *args, **kwargs):
...
+        authors = (force_list(get_value(json, 'authors.full_name')))
+        recid = get_value(json, 'self.$ref')
```
Comment (jmartinm) [inline inspirehep/modules/records/receivers.py]: its not really a recid (which is the number we use in the URL). Instead call it `ref` or `json_ref`

**Comment by jmartinm** [comment]
As discussed in person, besides the above comments, the only thing that would be nice to add is the validation of `page_start` and `page_end`

File: inspirehep/modules/literaturesuggest/forms.py [hunk 25]
```diff
@@ -183,6 +187,31 @@ class AuthorInlineForm(INSPIREForm):
     )
 
 
+class IsbnInlineForm(INSPIREForm):
```
Comment (Glignos) [inline inspirehep/modules/literaturesuggest/forms.py]: No we removed it from the submission page.

File: inspirehep/modules/records/receivers.py (hunk 23 above)
Comment (Glignos) [inline inspirehep/modules/records/receivers.py]: The purpose of this is to add these values to the autocompletion but in the payload as well so that I can fill the fields afterwards.

File: inspirehep/modules/literaturesuggest/normalizers.py [hunk 26]
```diff
@@ -75,9 +92,21 @@ def split_page_range_article_id(obj, formdata):
...
         formdata['journal_title'] = _normalize_journal_title(formdata['journal_title'])
+    return formdata
+
 
+def find_book_id(obj, formdata):
+    if formdata.get('type_of_doc') == 'chapter':
+        if not formdata.get('parent_book'):
+            result = check_book_existence(formdata.get('book_title'))
+            if result.rowcount == 1:
+                for row in result:
```
Comment (Glignos) [inline inspirehep/modules/literaturesuggest/normalizers.py]: The result does not support indexing so I am doing it this way for that reason.

File: inspirehep/modules/records/receivers.py (hunk 22 above)
Comment (Glignos) [inline inspirehep/modules/records/receivers.py]: Removing the force_list leads to an error in this case

Comment (jmartinm) [inline inspirehep/modules/records/receivers.py]: which one?

File: inspirehep/modules/records/receivers.py (hunk 23 above)
Comment (jmartinm) [inline inspirehep/modules/records/receivers.py]: What I mean is that you are running the same statements two times, you should write it liKe:

```python
if 'book' in force_list(json.get('document_type')):
        result = json.get('bookautocomplete', [])
        titles = force_list(get_value(json, 'titles.title'))
        authors = (force_list(get_value(json, 'authors.full_name')))
        result.extend(titles)
        result.extend(authors)
        ref = get_value(json, 'self.$ref')
        json.update({'bookautocomplete': {
                        'input': result,
                        'payload': {
                            'title': titles,
                            'authors': authors,
                            'id': ref
                            }
                        }
                     })

```

**Comment by jmartinm** [comment]
For the commit log message, please break it at ~80 chars

```
* Added 2 new fields(book,chapter) in the document type in the
   literature submission. (closes #861)
```

File: inspirehep/modules/records/receivers.py [hunk 27]
```diff
@@ -67,6 +67,27 @@ def enhance_record(sender, json, *args, **kwargs):
     populate_abstract_source_suggest(sender, json, *args, **kwargs)
     populate_title_suggest(sender, json, *args, **kwargs)
     after_record_enhanced.send(json)
+    add_book_autocomplete(sender, json, *args, **kwargs)
+    after_record_enhanced.send(json)
```
Comment (jacquerie) [inline inspirehep/modules/records/receivers.py]: Why the repetition from two lines above?

Comment (Glignos) [inline inspirehep/modules/records/receivers.py]: Fixed that.

File: inspirehep/bat/pages/create_literature.py [hunk 28]
```diff
@@ -70,6 +70,64 @@ def _submit_thesis():
...
+            'The INSPIRE staff will review it and your changes will be added '
+            'to INSPIRE.'
+        ) in WebDriverWait(Arsenic(), 10).until(
+            EC.visibility_of_element_located(
+                (
+                    By.XPATH,
+                    '(//div[@class="alert alert-success alert-form-success"])',
+                )
+            )
+        ).text
```
Comment (jacquerie) [inline inspirehep/bat/pages/create_literature.py]: This will have to modified when merging https://github.com/inspirehep/inspire-next/pull/2517.

File: inspirehep/bat/pages/create_literature.py [hunk 29]
```diff
@@ -70,6 +70,64 @@ def _submit_thesis():
...
+            'The INSPIRE staff will review it and your changes will be added '
+            'to INSPIRE.'
+        ) in WebDriverWait(Arsenic(), 10).until(
+            EC.visibility_of_element_located(
+                (
+                    By.XPATH,
+                    '(//div[@class="alert alert-warning alert-form-warning"])',
+                )
+            )
+        ).text
```
Comment (jacquerie) [inline inspirehep/bat/pages/create_literature.py]: This as well.

File: inspirehep/modules/literaturesuggest/forms.py [hunk 30]
```diff
@@ -242,22 +246,16 @@ class LiteratureForm(INSPIREForm):
...
     types_of_doc = [("article", _("Article/Conference paper")),
-                    ("thesis", _("Thesis"))]
-
-    # ("chapter", _("Book Chapter")),
-    # ("book", _("Book")),
-    # ("proceedings", _("Proceedings"))]
+                    ("thesis", _("Thesis")),
+                    ('book', _('Book')),
+                    ('chapter', _('Book chapter'))]
+    # ("proceedings", _("Proceedings"))
```
Comment (jacquerie) [inline inspirehep/modules/literaturesuggest/forms.py]: Try not to leave commented code without a comment or a pointer to an issue that explains what's missing here.

File: inspirehep/modules/literaturesuggest/normalizers.py [hunk 31]
```diff
@@ -31,6 +31,22 @@
     normalize_journal_title as _normalize_journal_title,
 )
 from inspirehep.utils.pubnote import split_page_artid
+from invenio_db import db
+from sqlalchemy import text
+
+
+def check_journal_existance(title):
+    query = text("SELECT  id FROM records_metadata as r, json_array_elements_text(r.json -> '_collections') as elem,json_array_elements(r.json-> 'journal_titles') as titles\
+    WHERE 'Journals' = elem AND titles->>'title' =:j_title;").bindparams(j_title=title)
```
Comment (jacquerie) [inline inspirehep/modules/literaturesuggest/normalizers.py]: For ease of reading these queries should be formatted properly.
```SQL
SELECT id
FROM
    records_metadata AS r,
    json_array_elements_text(r.json -> '_collections') as elem,
    json_array_elements(r.json-> 'journal_titles') as titles
WHERE
    'Journals' = elem AND
    titles->>'title' = :j_title;
```

File: inspirehep/modules/forms/static/scss/forms/form.scss [hunk 32]
```diff
@@ -31,56 +31,66 @@
 }
 
 .ui-autocomplete-loading {
-     background: white url('/static/images/loading.gif') right top no-repeat;
+  background: white url('/static/images/loading.gif') right top no-repeat;
```
Comment (jacquerie) [inline inspirehep/modules/forms/static/scss/forms/form.scss]: Why the switch to 2 spaces for indentation? The rest of the CSS codebase seems to use 4...

File: tests/integration/utils/test_normalizers.py [hunk 33]
```diff
@@ -23,6 +23,11 @@
 from __future__ import absolute_import, division, print_function
 
 from inspirehep.utils.normalizers import normalize_journal_title
+from inspirehep.modules.literaturesuggest import normalizers
+
+
+class MockObj:
+    pass
```
Comment (jacquerie) [inline tests/integration/utils/test_normalizers.py]: We already have this guy, no need to define him again.

File: tests/unit/literaturesuggest/test_literaturesuggest_normalizers.py [hunk 34]
```diff
@@ -38,9 +38,9 @@ def test_split_page_range_article_id():
 
     expected = {
         'artid': None,
-        'page_end': '890',
+        'end_page': '890',
         'page_range_article_id': '789-890',
-        'page_start': '789',
+        'start_page': '789',
```
Comment (jacquerie) [inline tests/unit/literaturesuggest/test_literaturesuggest_normalizers.py]: Uhm, where we producing invalid records before? Sounds like something that our tests should have caught...

Comment (Glignos) [inline tests/unit/literaturesuggest/test_literaturesuggest_normalizers.py]: No i did I renaming on some fields.

File: tests/integration/utils/test_normalizers.py (hunk 33 above)
Comment (Glignos) [inline tests/integration/utils/test_normalizers.py]: There was this issue that this class was in the unit tests and It couldnt be imported

File: inspirehep/modules/forms/static/scss/forms/form.scss (hunk 32 above)
Comment (Glignos) [inline inspirehep/modules/forms/static/scss/forms/form.scss]: This actually happened because of the autoformatting on my editor. I fixed most of the files but I missed this one.Should I revert it?
